### ISSUE TITLE
RRule::from_str panics at empty strings.

### ISSUE THREAD

**Issue by akhramov** [fmeringdal/rust_rrule#17]
Hey there :)

I've noticed that `FromStr` implementation panics on empty value:

```rust
RRule::from_str("");
```

results in

```
thread 'tokio-runtime-worker' panicked at 'attempt to subtract with overflow', /home/akhramov/.cargo/registry/src/github.com-1ecc6299db9ec823/rrule-0.5.7/src/rrulestr.rs:567:44
```

What happens in this case is rrule_vals ends up being empty, i.e. of zero length:

https://github.com/fmeringdal/rust_rrule/blob/7c8d64b028b3c7c1b55b55ad088ecbcb48d8bdc5/src/rrulestr.rs#L394-L395

Then we attempt to subtract 1 from zero, which is invalid for `usize`, an unsigned type

https://github.com/fmeringdal/rust_rrule/blob/7c8d64b028b3c7c1b55b55ad088ecbcb48d8bdc5/src/rrulestr.rs#L567

All in all, this library saved me tons of time and I can't express how grateful I am. Thank you!

**Comment by fmeringdal** [comment]
Hi, thanks for reporting the issue.

>  All in all, this library saved me tons of time and I can't express how grateful I am. Thank you!

It makes me very happy to hear that you liked this library :)

I am planning to release a new version within a week a so, and will remember to add a fix for this issue to that release. Will keep you updated here when it is released.

Please let me know if you find more bugs!

**Comment by fmeringdal** [comment]
This should now be fixed in version 0.5.8
